experimental/dds/tree2/src/test/feature-libraries/optional-field/optionalChangeRebaser.spec.ts

```diff
@@ -232,32 +307,56 @@ const generateChildStates: ChildStateGenerator<string | undefined, OptionalChang
 
 	for (const value of ["A", "B"]) {
 		const setIntention = mintIntention();
+		const [fill, detach] = [mintId(), mintId()];
 		// Using length of the input context guarantees set operations generated at different times also have different
-		// values, which should tend to be easier to debug
-		const newContents = `${value},${inputContext.length}`;
+		// values, which should tend to be easier to debug.
+		// This also makes the logic to determine intentions simpler.
+		const newContents = `${value},${edits.length}`;
 		yield {
 			content: newContents,
 			mostRecentEdit: {
 				changeset: tagChange(
-					OptionalChange.set(newContents, state.content === undefined),
+					OptionalChange.set(newContents, state.content === undefined, {
+						fill,
+						detach,
+					}),
 					tagFromIntention(setIntention),
 				),
 				intention: setIntention,
-				description: `Set${value},${inputContext.length}`,
+				description: `Set${value},${edits.length}`,
```

Comment:
It might get confusing if one changes `newContents` without changing `description` or vice-versa.
```suggestion
				description: `Set${newContents}`,
```